src/client/common/application/customEditorService.ts

```diff
@@ -65,24 +62,17 @@ export class CustomEditorService implements ICustomEditorService {
                 viewType: ViewType,
                 filenamePattern: '*.ipynb'
             });
-            await Promise.all([
-                this.extensionContext.globalState.update(EditorAssociationUpdatedKey, true),
-                settings.update('editorAssociations', editorAssociations, vscode.ConfigurationTarget.Global)
-            ]);
+            await settings.update('editorAssociations', editorAssociations, vscode.ConfigurationTarget.Global);
```

Comment:
This is a bit tricky, as we don't have an exact lockdown from the associated customer issue. However this change should prevent getting incorrect editor associations if you copy around settings files.

This seems like the right fix regardless as the settings file isn't scoped the same as the extension global state so one can't be used to guard the other.

Reply:
Makes perfect sense.